### PR TITLE
Remove urlSearch mixin

### PR DESCRIPTION
**Pull request by spen**
This PR removes the `urlSearch` mixin, replacing it's remaining implementations with the `urlSearch` higher-order-component.

#### people-section-nav/index.jsx
- Go to http://calypso.localhost:3000/people/team/your-site.wordpress.com
- Click the search icon at the top & Start typing
  - Note that the people in this section start to filter
  - Also note that the url will update with your search term like so: `?s={ your search term }`

#### plugins/plugins-browser/index.jsx
- Go to http://calypso.localhost:3000/plugins/your-site.wordpress.com
- Click the search icon at the top & Start typing
  - Note that the plugins available should start to filter down
  - Also note that the url will update with your search term like so: `?s={ your search term }`

#### plugins/main.jsx
- Go to http://calypso.localhost:3000/plugins/manage/some-jetpack-site.com
- Click the search icon at the top & Start typing
  - Note that your sites installed plugins should start to filter
  - Also note that the url will update with your search term like so: `?s={ your search term }`

#### post-type-filter/index.jsx

- Go to http://calypso.localhost:3000/posts/your-site.wordpress.com
- Click the search icon at the top & Start typing
  - Note that your posts should start to filter
  - Also note that the url will update with your search term like so: `?s={ your search term }`

### CODE REVIEW

**Comment by lsinger** [comment]
Rebased and fixed a small issue in f99f293. 

I guess it's unrelated to this PR, but note how searching for `Jetpack` is ... interesting:

plugin-search

(Plugins)

plugin-search-jetpack

(Plugins / Manage)

**Comment by spen** [comment]
That's quite bizarre! It'll be worth doing a bit of 🕵️work and looking for the cause here.
Thanks for the rebase too! <3

**Comment by spen** [comment]
@samouri @ockham @jsnajdr - Hope y'all don't mind the review req, feel free to ignore if you're busy!

File: client/lib/url-search/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+);
+```
+
+Then in the component file, enhance with `urlSearch`:
+
+```js
+/**
+ * Internal dependencies
+ */
+import urLSearch from 'lib/url-search';
```
Comment (samouri) [inline client/lib/url-search/README.md]: typo: `urLSearch` --> `urlSearch`

File: client/my-sites/plugins/main.jsx [hunk 2]
```diff
@@ -47,21 +46,20 @@ import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/u
 import HeaderButton from 'components/header-button';
 import { isEnabled } from 'config';
 
-const PluginsMain = createReactClass( {
-	displayName: 'PluginsMain',
-	mixins: [ URLSearch ],
+export class PluginsMain extends Component {
+	constructor( props ) {
```
Comment (samouri) [inline client/my-sites/plugins/main.jsx]: nit: we don't need a constructor. it is valid to write code that rely upon the [class property transforms](https://babeljs.io/docs/plugins/transform-class-properties/) like this:

```js
export class PluginsMain extends Component {
  state = this.getPluginsState( this.props );
}
```

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
 /** @format */
+/* eslint-disable react/prefer-es6-class */
```
Comment (samouri) [inline client/my-sites/plugins/plugins-browser/index.jsx]: This PR removes the only `mixin` right?
I think that means we can update the `createReactClass` call to a plain es6 class

File: client/my-sites/post-type-filter/index.jsx [hunk 4]
```diff
@@ -30,11 +29,8 @@ import AuthorSegmented from './author-segmented';
 import Button from 'components/button';
 import Gridicon from 'gridicons';
 
-const PostTypeFilter = createReactClass( {
-	displayName: 'PostTypeFilter',
-	mixins: [ UrlSearch ],
-
-	propTypes: {
+export class PostTypeFilter extends Component {
```
Comment (samouri) [inline client/my-sites/post-type-filter/index.jsx]: missing a `static displayName = 'PostTypeFilter'`

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 3 above)
Comment (ockham) [reply client/my-sites/plugins/plugins-browser/index.jsx]: What @samouri said. Probably easiest to just run the file through the codemod 🙂

**Comment by samouri** [comment]
Thanks for the awesome work @spen! I mostly had small nits and would be happy to help with updating the class to es6 if you'd like

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 3 above)
Comment (spen) [reply client/my-sites/plugins/plugins-browser/index.jsx]: This is a bit of a hangover from when I originally opened this. That was before the infinite scroll mixin was replaced & removed from this file. I do have a PR open #23695 (somethings gone awry with the rebasing so all commits are shown there).

It does make sense at this point to just apply that commit to this PR though :)

Comment (spen) [reply client/my-sites/plugins/plugins-browser/index.jsx]: > Probably easiest to just run the file through the codemod

I thought that but I don't think it picks up cases of `createReactClass` - I'm a little rusty on that front though so might have just been doing something wrong.

Comment (ockham) [reply client/my-sites/plugins/plugins-browser/index.jsx]: No, you're right. You'd need to change it back to `createClass` (and `import { createClass } from 'react';`) first in order for the codemod to pick it up.

File: client/my-sites/post-type-filter/index.jsx (hunk 4 above)
Comment (spen) [reply client/my-sites/post-type-filter/index.jsx]: Fixed up and folded back in via a rebase :)

I always thought we inferred the `displayName` from the declaration. 
Turns out that we don't, so I'm not sure where I got this from.
Anyway, we do get `name` (`this.constructor.name`) from the declaration, which is enough to give us the info we need for debugging etc.:

```
The above error occurred in the <PluginsBrowser> component:
    in PluginsBrowser (created by Localized(PluginsBrowser))
    in Localized(PluginsBrowser) (created by UrlSearch(Localized(PluginsBrowser)))
```

(From `PluginsBrowser` before adding `static displayName ...`)

File: client/my-sites/plugins/main.jsx (hunk 2 above)
Comment (spen) [reply client/my-sites/plugins/main.jsx]: Good shout! :)

**Comment by spen** [comment]
Thanks for the 👀s! I think that's everything addressed - I also noticed that there were some unbound methods being passed to different contexts in `plugins/main.jsx` so I fixed them up too :)

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 3 above)
Comment (ockham) [reply client/my-sites/plugins/plugins-browser/index.jsx]: We can drop the `/* eslint-disable react/prefer-es6-class */` line now, can't we? 🎉

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 5]
```diff
@@ -47,19 +47,32 @@ import Banner from 'components/banner';
...
+
+	constructor( props ) {
+		super( props );
+
+		const pluginsList = this.getPluginsLists( this.props.search );
+
+		this.state = {
+			...pluginsList,
+		};
+	}
```
Comment (ockham) [inline client/my-sites/plugins/plugins-browser/index.jsx]: Can we ditch the constructor and use the property initializer shorthand for `state` here?

```jsx
export class PluginsBrowser extends Component {
	static displayName = 'PluginsBrowser';

	state = ...this.getPluginsLists( this.props.search );

	/*...*/
}
```

Comment (spen) [reply client/my-sites/plugins/plugins-browser/index.jsx]: Sorted, cheers @ockham :)

**Comment by samouri** [comment]
Sorry for the delay!
Code here LGTM. 

I'll rebase, fix up the merge conflicts, test all the modified components, and then 🚢

**Comment by spen** [comment]
NP! Thanks heaps for the rebase fix up and thanks again for reviewing! :)

**Pull request merged**